core_lib/src/util/painterutils.h

```diff
@@ -0,0 +1,19 @@
+#ifndef PAINTERUTILS_H
```

Comment:
util.h adds several headers though that include GUI related stuff like QApplication and QSpinBox, i'd like to avoid those if possible.

though of course painting in general is part of QT GUI... 🤔 still though, the addition of more headers makes me think it shouldn't have that coupling.

In the future it maybe we can create a basePainter to handle this instead but right now I think a standalone header is fine.